### PR TITLE
MULE-10991 Change default processing strategy to treat one-way and re…

### PR DESCRIPTION
**Pull request by dfeist**
…quest-response equally.

### CODE REVIEW

File: tests/functional/src/main/java/org/mule/functional/junit4/FlowRunner.java [hunk 1]
```diff
@@ -113,18 +117,47 @@ public Event runNoVerify() throws Exception {
...
+   *
+   * Dispatch behaves differently to {@link FlowRunner#run()} in that it does execute the flow in a seperate thread and does not
+   * propagate any exceptions to the test case or return a result.
+   *
+   */
+  public void dispatch() {
+    Flow flow = (Flow) getFlowConstruct();
+    asyncScheduler.execute(() -> {
+      try {
+        txExecutionTemplate.execute(() -> {
```
Comment (elrodro83) [inline tests/functional/src/main/java/org/mule/functional/junit4/FlowRunner.java]: this body is the same as in runAndVerify, extract to a single location

File: tests/functional/src/main/java/org/mule/functional/junit4/FlowRunner.java [hunk 2]
```diff
@@ -113,18 +117,47 @@ public Event runNoVerify() throws Exception {
...
+          if (nonBlocking) {
+            return processAsStreamAndBlock(getOrBuildEvent(), flow);
+          } else {
+            return flow.process(getOrBuildEvent());
+          }
+        });
+      } catch (Exception e) {
+        // Ignore
```
Comment (elrodro83) [inline tests/functional/src/main/java/org/mule/functional/junit4/FlowRunner.java]: should this at least be logged?

File: tests/integration-tranports/src/test/java/org/mule/test/construct/FlowUseCaseProcessingStrategyTestCase.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ public void testFileAutoDeleteAsyncStrategy() throws Exception {
     File tempFile = createTempFile("mule-file-test-async-");
     client.request("vm://exception", 5000);
 
-    assertFalse(tempFile.exists());
+    assertTrue(tempFile.exists());
```
Comment (elrodro83) [inline tests/integration-tranports/src/test/java/org/mule/test/construct/FlowUseCaseProcessingStrategyTestCase.java]: why?

File: tests/integration-tranports/src/test/java/org/mule/test/integration/exceptions/OnErrorPropagateTestCase.java [hunk 4]
```diff
@@ -278,7 +279,7 @@ public void testRollbackExceptionStrategyCatchMessageRedeliveryDespiteChoiceConf
 
   private void verifyFlow(String flowName, Object payload) throws InterruptedException {
     try {
-      flowRunner(flowName).withPayload(payload).asynchronously().run().getMessage();
+      flowRunner(flowName).withPayload(payload).withExchangePattern(ONE_WAY).run().getMessage();
```
Comment (elrodro83) [inline tests/integration-tranports/src/test/java/org/mule/test/integration/exceptions/OnErrorPropagateTestCase.java]: all other cases are being migrated to dispatch(), why not this one?

File: tests/integration/src/test/java/org/mule/shutdown/ExpiredShutdownTimeoutOneWayTestCase.java [hunk 5]
```diff
@@ -49,7 +52,7 @@ private void doShutDownTest(final String flowName) throws MuleException, Interru
       @Override
       public void run() {
         try {
-          flowRunner(flowName).withPayload(TEST_MESSAGE).asynchronously().run();
+          flowRunner(flowName).withPayload(TEST_MESSAGE).withExchangePattern(ONE_WAY).dispatch();
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/shutdown/ExpiredShutdownTimeoutOneWayTestCase.java]: all other cases are being migrated to dispatch(), why not this one?

File: tests/integration/src/test/java/org/mule/shutdown/ValidShutdownTimeoutOneWayTestCase.java [hunk 6]
```diff
@@ -55,7 +57,7 @@ private void doShutDownTest(final String payload, final String flowName) throws
       @Override
       public void run() {
         try {
-          flowRunner(flowName).withPayload(payload).asynchronously().run();
+          flowRunner(flowName).withPayload(payload).withExchangePattern(ONE_WAY).dispatch();
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/shutdown/ValidShutdownTimeoutOneWayTestCase.java]: all other cases are being migrated to dispatch(), why not this one?

File: tests/integration/src/test/java/org/mule/test/construct/FlowConfigurationFunctionalTestCase.java [hunk 7]
```diff
@@ -77,27 +79,27 @@ public void testFlowSynchronous() throws Exception {
...
     Thread thread = (Thread) message.getPayload().getValue();
     assertNotNull(thread);
-    assertEquals(Thread.currentThread(), thread);
+    assertEquals(currentThread(), thread);
   }
 
   @Test
   public void testFlowAynchronous() throws Exception {
-    flowRunner("asynchronousFlow").withPayload("0").asynchronously().run();
+    flowRunner("asynchronousFlow").withPayload("0").withExchangePattern(ONE_WAY).dispatch();
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/construct/FlowConfigurationFunctionalTestCase.java]: all other cases are being migrated to dispatch(), why not this one?

File: tests/integration/src/test/java/org/mule/test/construct/FlowConfigurationFunctionalTestCase.java [hunk 8]
```diff
@@ -395,7 +397,7 @@ public void testAsyncRequestResponseEndpoint() throws Exception {
   @Test
   public void testAsyncTransactionalEndpoint() throws Exception {
     Exception e = flowRunner("async-tx").withPayload("0").transactionally(ACTION_ALWAYS_BEGIN, new TestTransactionFactory())
-        .asynchronously().runExpectingException();
+        .withExchangePattern(ONE_WAY).runExpectingException();
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/construct/FlowConfigurationFunctionalTestCase.java]: all other cases are being migrated to dispatch(), why not this one?

File: tests/integration/src/test/java/org/mule/test/core/NonBlockingFullySupportedOneWayFunctionalTestCase.java [hunk 9]
```diff
@@ -63,7 +65,8 @@ public void filterAccepts() throws Exception {
 
   @Test
   public void filterRejects() throws Exception {
-    assertThat(flowRunner("filterRejects").withPayload(TEST_MESSAGE).asynchronously().nonBlocking().run(), is(nullValue()));
+    assertThat(flowRunner("filterRejects").withPayload(TEST_MESSAGE).nonBlocking().withExchangePattern(ONE_WAY).run(),
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/core/NonBlockingFullySupportedOneWayFunctionalTestCase.java]: all other cases are being migrated to dispatch(), why not this one?

File: tests/integration/src/test/java/org/mule/test/core/NonBlockingFullySupportedOneWayFunctionalTestCase.java [hunk 10]
```diff
@@ -83,13 +86,14 @@ public void filterBeforeNonBlockingAccepts() throws Exception {
 
   @Test
   public void filterBeforeNonBlockingRejects() throws Exception {
-    assertThat(flowRunner("filterBeforeNonBlockingRejects").withPayload(TEST_MESSAGE).asynchronously().nonBlocking().run(),
+    assertThat(flowRunner("filterBeforeNonBlockingRejects").withPayload(TEST_MESSAGE).nonBlocking().run(),
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/core/NonBlockingFullySupportedOneWayFunctionalTestCase.java]: shoulnd't this be dispatch()?

File: tests/integration/src/test/java/org/mule/test/integration/messaging/meps/InOnlyOutOnlyTestCase.java [hunk 11]
```diff
@@ -27,10 +27,10 @@ protected String getConfigFile() {
   public void testExchange() throws Exception {
     MuleClient client = muleContext.getClient();
 
-    FlowRunner baseRunner = flowRunner("In-Only_Out-Only-Service").withPayload("some data").asynchronously();
+    FlowRunner baseRunner = flowRunner("In-Only_Out-Only-Service").withPayload("some data");
     baseRunner.run();
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/integration/messaging/meps/InOnlyOutOnlyTestCase.java]: shouldn't thid change to dispatch() also?

File: tests/integration/src/test/java/org/mule/test/integration/routing/MessageChunkingTestCase.java [hunk 12]
```diff
@@ -87,16 +85,13 @@ public void onNotification(ServerNotification notification) {
     flowExecutionListener.addListener(source -> messagePartsCount.getAndIncrement());
 
     MuleClient client = muleContext.getClient();
-    flowRunner("ObjectReceiver").withPayload(simpleSerializableObject).asynchronously().run();
-    // Wait for the message to be received and tested (in the listener above)
-    assertTrue(chunkingReceiverLatch.await(20L, TimeUnit.SECONDS));
+    flowRunner("ObjectReceiver").withPayload(simpleSerializableObject).run();
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/integration/routing/MessageChunkingTestCase.java]: are the changes in this class correct? why remove the latches and meke this synchronous?

File: tests/integration/src/test/java/org/mule/test/integration/transaction/xa/XaTransactionsPersistentQueueTestCase.java [hunk 13]
```diff
@@ -29,7 +31,7 @@ protected String getConfigFile() {
   public void testOutboundRouterTransactions() throws Exception {
     MuleClient client = muleContext.getClient();
 
-    flowRunner("XaTestService").withPayload(TEST_MESSAGE).asynchronously()
+    flowRunner("XaTestService").withPayload(TEST_MESSAGE).withExchangePattern(ONE_WAY)
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/integration/transaction/xa/XaTransactionsPersistentQueueTestCase.java]: all other cases are being migrated to dispatch(), why not this one?

File: transports/file/src/test/java/org/mule/compatibility/transport/file/FileExceptionStrategyFunctionalTestCase.java [hunk 14]
```diff
@@ -83,7 +83,7 @@ public void testCopyFile() throws Exception {
     latch.await(2000l, MILLISECONDS);
     flow.stop();
     File outputFile = getFileInsideWorkingDirectory("temp/output-directory/" + inputFile.getName());
-    assertThat(inputFile.exists(), is(false));
+    assertThat(inputFile.exists(), is(true));
```
Comment (elrodro83) [inline transports/file/src/test/java/org/mule/compatibility/transport/file/FileExceptionStrategyFunctionalTestCase.java]: why?

File: transports/file/src/test/java/org/mule/compatibility/transport/file/FileExceptionStrategyFunctionalTestCase.java [hunk 15]
```diff
@@ -97,7 +97,7 @@ public void testCopyFileWithWorkDir() throws Exception {
     flow.stop();
     File outputFile = getFileInsideWorkingDirectory("temp/output-directory/" + inputFile.getName());
     File workDirFile = getFileInsideWorkingDirectory(FILE_WORKING_DIRECTORY_FOLDER + File.separator + inputFile.getName());
-    assertThat(inputFile.exists(), is(false));
+    assertThat(inputFile.exists(), is(true));
```
Comment (elrodro83) [inline transports/file/src/test/java/org/mule/compatibility/transport/file/FileExceptionStrategyFunctionalTestCase.java]: why?

File: transports/file/src/test/java/org/mule/compatibility/transport/file/reliability/InboundMessageLossTestCase.java [hunk 16]
```diff
@@ -75,23 +77,17 @@ public String describeFailure() {
...
 
-      @Override
-      public String describeFailure() {
-        return "File should be gone";
-      }
-    });
+    latch.await(RECEIVE_TIMEOUT, MILLISECONDS);
+    stopIfNeeded(muleContext.getRegistry().lookupFlowConstruct("ComponentException"));
+    muleContext.getRegistry().lookupObject(FileConnector.class).stop();
+    assertThat(file.exists(), is(true));
```
Comment (elrodro83) [inline transports/file/src/test/java/org/mule/compatibility/transport/file/reliability/InboundMessageLossTestCase.java]: why?

File: transports/file/src/test/java/org/mule/compatibility/transport/file/reliability/InboundMessageLossTestCase.java [hunk 17]
```diff
@@ -173,46 +163,34 @@ public String describeFailure() {
...
-
-      @Override
-      public String describeFailure() {
-        return "File should be gone";
-      }
-    });
+    latch.await(RECEIVE_TIMEOUT, MILLISECONDS);
+    stopIfNeeded(muleContext.getRegistry().lookupFlowConstruct("RouterException"));
+    muleContext.getRegistry().lookupObject(FileConnector.class).stop();
+    assertThat(file.exists(), is(true));
```
Comment (elrodro83) [inline transports/file/src/test/java/org/mule/compatibility/transport/file/reliability/InboundMessageLossTestCase.java]: why?

File: transports/file/src/test/java/org/mule/compatibility/transport/file/reliability/InboundMessageLossTestCase.java [hunk 18]
```diff
@@ -173,46 +163,34 @@ public String describeFailure() {
...
 
-      @Override
-      public String describeFailure() {
-        return "File should be gone";
-      }
-    });
+    latch.await(RECEIVE_TIMEOUT, MILLISECONDS);
+    stopIfNeeded(muleContext.getRegistry().lookupFlowConstruct("TransformerException"));
+    muleContext.getRegistry().lookupObject(FileConnector.class).stop();
+    assertThat(file.exists(), is(true));
```
Comment (elrodro83) [inline transports/file/src/test/java/org/mule/compatibility/transport/file/reliability/InboundMessageLossTestCase.java]: why?

File: transports/file/src/test/resources/reliability/inbound-message-loss-flow.xml [hunk 19]
```diff
@@ -30,7 +30,7 @@
         <flow-ref name="failingFlow"/>
     </flow>
     <flow name="failingFlow" processingStrategy="synchronous">
-        <test:component/>
+        <test:component throwException="true"/>
```
Comment (elrodro83) [inline transports/file/src/test/resources/reliability/inbound-message-loss-flow.xml]: why is this change needed in the test?

File: transports/http/src/test/java/org/mule/compatibility/transport/http/reliability/InboundMessageLossAsynchTestCase.java [hunk 20]
```diff
@@ -48,26 +52,21 @@ public void testHandledTransformerException() throws Exception {
   @Override
   public void testComponentException() throws Exception {
     HttpMethodBase request = createRequest(getBaseUri() + "/componentException");
-    int status = httpClient.executeMethod(request);
-    // Component exception occurs after the SEDA queue for an asynchronous request, so from the client's
-    // perspective, the message has been delivered successfully.
-    assertEquals(HttpConstants.SC_OK, status);
+    assertThat(httpClient.executeMethod(request), equalTo(SC_INTERNAL_SERVER_ERROR));
```
Comment (elrodro83) [inline transports/http/src/test/java/org/mule/compatibility/transport/http/reliability/InboundMessageLossAsynchTestCase.java]: why?

File: transports/http/src/test/java/org/mule/compatibility/transport/http/reliability/InboundMessageLossAsynchTestCase.java [hunk 21]
```diff
@@ -48,26 +52,21 @@ public void testHandledTransformerException() throws Exception {
...
   }
 
   @Test
   @Override
   public void testTransformerException() throws Exception {
     HttpMethodBase request = createRequest(getBaseUri() + "/transformerException");
-    int status = httpClient.executeMethod(request);
-    assertEquals(HttpConstants.SC_OK, status);
+    assertThat(httpClient.executeMethod(request), equalTo(SC_INTERNAL_SERVER_ERROR));
```
Comment (elrodro83) [inline transports/http/src/test/java/org/mule/compatibility/transport/http/reliability/InboundMessageLossAsynchTestCase.java]: why?

File: transports/http/src/test/java/org/mule/compatibility/transport/http/reliability/InboundMessageLossAsynchTestCase.java [hunk 22]
```diff
@@ -48,26 +52,21 @@ public void testHandledTransformerException() throws Exception {
...
   }
 
   @Test
   @Override
   public void testRouterException() throws Exception {
     HttpMethodBase request = createRequest(getBaseUri() + "/routerException");
-    int status = httpClient.executeMethod(request);
-    assertEquals(HttpConstants.SC_OK, status);
+    assertThat(httpClient.executeMethod(request), equalTo(SC_INTERNAL_SERVER_ERROR));
```
Comment (elrodro83) [inline transports/http/src/test/java/org/mule/compatibility/transport/http/reliability/InboundMessageLossAsynchTestCase.java]: why?

File: transports/jms/src/test/java/org/mule/compatibility/transport/jms/reliability/InboundMessageLossTestCase.java [hunk 23]
```diff
@@ -66,44 +66,33 @@ public void testNoException() throws Exception {
...
   @Test
   public void testTransformerException() throws Exception {
     putMessageOnQueue("transformerException");
-
-    // Exception occurs after the SEDA queue for an asynchronous request, so from the client's
-    // perspective, the message has been delivered successfully.
-    // Note that this behavior is different from services because the exception occurs before
-    // the SEDA queue for services.
-    assertFalse("Message should not have been redelivered", messageRedelivered.await(latchTimeout, TimeUnit.MILLISECONDS));
+    assertThat("Message should not have been redelivered", messageRedelivered.await(latchTimeout, MILLISECONDS), is(true));
```
Comment (elrodro83) [inline transports/jms/src/test/java/org/mule/compatibility/transport/jms/reliability/InboundMessageLossTestCase.java]: why?

File: transports/jms/src/test/java/org/mule/compatibility/transport/jms/reliability/InboundMessageLossTestCase.java [hunk 24]
```diff
@@ -66,44 +66,33 @@ public void testNoException() throws Exception {
...
 
   @Test
   public void testRouterException() throws Exception {
-    // Exception occurs after the SEDA queue for an asynchronous request, so from the client's
-    // perspective, the message has been delivered successfully.
-    // Note that this behavior is different from services because the exception occurs before
-    // the SEDA queue for services.
-    assertFalse("Message should not have been redelivered", messageRedelivered.await(latchTimeout, TimeUnit.MILLISECONDS));
+    putMessageOnQueue("routerException");
+    assertThat("Message should not have been redelivered", messageRedelivered.await(latchTimeout, MILLISECONDS), is(true));
```
Comment (elrodro83) [inline transports/jms/src/test/java/org/mule/compatibility/transport/jms/reliability/InboundMessageLossTestCase.java]: why?

File: transports/jms/src/test/java/org/mule/compatibility/transport/jms/reliability/InboundMessageLossTestCase.java [hunk 25]
```diff
@@ -66,44 +66,33 @@ public void testNoException() throws Exception {
...
   }
 
   @Test
   public void testComponentException() throws Exception {
     putMessageOnQueue("componentException");
-
-    // Exception occurs after the SEDA queue for an asynchronous request, so from the client's
-    // perspective, the message has been delivered successfully.
-    assertFalse("Message should not have been redelivered", messageRedelivered.await(latchTimeout, TimeUnit.MILLISECONDS));
+    assertThat("Message should not have been redelivered", messageRedelivered.await(latchTimeout, MILLISECONDS), is(true));
```
Comment (elrodro83) [inline transports/jms/src/test/java/org/mule/compatibility/transport/jms/reliability/InboundMessageLossTestCase.java]: why?

File: core-tests/src/test/java/org/mule/runtime/core/construct/FlowTestCase.java [hunk 26]
```diff
@@ -124,11 +127,8 @@ public void testProcessOneWayEndpoint() throws Exception {
 
       @Override
       protected boolean test() throws Exception {
-        // While a SedaService returns null, a Flow echos the request when there is async hand-off
-        assertEquals(event.getMessage(), response.getMessage());
-
-        assertEquals("helloabc", sensingMessageProcessor.event.getMessageAsString(muleContext));
-        assertNotSame(Thread.currentThread(), sensingMessageProcessor.event.getVariable("thread").getValue());
+        assertThat(response.getMessageAsString(muleContext), equalTo(TEST_PAYLOAD + "abcdef"));
+        assertThat(sensingMessageProcessor.event.getVariable("thread").getValue(), not(sameInstance(currentThread())));
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/construct/FlowTestCase.java]: Is there an easy way to use what you implemented from test case in assertions?

File: tests/integration-tranports/src/test/java/org/mule/test/integration/exceptions/OnErrorPropagateTestCase.java (hunk 4 above)
Comment (dfeist) [inline tests/integration-tranports/src/test/java/org/mule/test/integration/exceptions/OnErrorPropagateTestCase.java]: This doesn't need dispatch or ONE_WAY, removing both.

File: tests/integration/src/test/java/org/mule/shutdown/ValidShutdownTimeoutOneWayTestCase.java (hunk 6 above)
Comment (dfeist) [inline tests/integration/src/test/java/org/mule/shutdown/ValidShutdownTimeoutOneWayTestCase.java]: same here.

File: tests/integration/src/test/java/org/mule/test/construct/FlowConfigurationFunctionalTestCase.java (hunk 7 above)
Comment (dfeist) [inline tests/integration/src/test/java/org/mule/test/construct/FlowConfigurationFunctionalTestCase.java]: Switched to used `run()`, as test doesn't depend on dispatch.  I'll review other uses of dispatch to see if any can be changed.

File: tests/integration/src/test/java/org/mule/test/core/NonBlockingFullySupportedOneWayFunctionalTestCase.java (hunk 9 above)
Comment (dfeist) [inline tests/integration/src/test/java/org/mule/test/core/NonBlockingFullySupportedOneWayFunctionalTestCase.java]: This is fine, needs ONE_WAY but not dispatch.

File: tests/integration/src/test/java/org/mule/test/core/NonBlockingFullySupportedOneWayFunctionalTestCase.java (hunk 10 above)
Comment (dfeist) [inline tests/integration/src/test/java/org/mule/test/core/NonBlockingFullySupportedOneWayFunctionalTestCase.java]: No, it's needs ONE_WAY only.

File: tests/integration-tranports/src/test/java/org/mule/properties/SessionPropertiesTestCase.java [hunk 27]
```diff
@@ -58,16 +58,14 @@ public void flowRefSessionPropertyPropagation() throws Exception {
     assertThat(result.getSession().getProperty("keyNonSerializable"), is(nonSerializable));
   }
 
-
-
   @Test
   public void defaultExceptionStrategy() throws Exception {
-    flowRunner("defaultExceptionStrategy").asynchronously().run();
+    flowRunner("defaultExceptionStrategy").dispatch();
```
Comment (elrodro83) [inline tests/integration-tranports/src/test/java/org/mule/properties/SessionPropertiesTestCase.java]: should this go back to run? review other dispatch() invocations.

File: core-tests/src/test/java/org/mule/runtime/core/construct/FlowTestCase.java [hunk 28]
```diff
@@ -124,11 +127,8 @@ public void testProcessOneWayEndpoint() throws Exception {
 
       @Override
       protected boolean test() throws Exception {
-        // While a SedaService returns null, a Flow echos the request when there is async hand-off
-        assertEquals(event.getMessage(), response.getMessage());
-
-        assertEquals("helloabc", sensingMessageProcessor.event.getMessageAsString(muleContext));
-        assertNotSame(Thread.currentThread(), sensingMessageProcessor.event.getVariable("thread").getValue());
+        assertThat(response.getMessageAsString(muleContext), equalTo(TEST_PAYLOAD + "abcdef"));
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/construct/FlowTestCase.java]: Because response is now returned, rather than echoing input.

File: tests/integration/src/test/java/org/mule/test/integration/routing/MessageChunkingTestCase.java (hunk 12 above)
Comment (dfeist) [inline tests/integration/src/test/java/org/mule/test/integration/routing/MessageChunkingTestCase.java]: latch is not needed  as run() won't complete until flow is executed now.

File: transports/file/src/test/resources/reliability/inbound-message-loss-flow.xml (hunk 19 above)
Comment (dfeist) [inline transports/file/src/test/resources/reliability/inbound-message-loss-flow.xml]: It replaces programming introduction of exception via FunctionalTestComponent callback..

File: tests/integration-tranports/src/test/java/org/mule/properties/SessionPropertiesTestCase.java (hunk 27 above)
Comment (dfeist) [inline tests/integration-tranports/src/test/java/org/mule/properties/SessionPropertiesTestCase.java]: This fails with run, as exceptions are propagated to test case with run.

File: tests/integration/src/test/java/org/mule/test/core/routing/outbound/AggregationTimeoutTestCase.java [hunk 29]
```diff
@@ -44,7 +45,10 @@ public void timeoutsAggregationWithPersistentStore() throws Exception {
 
     try {
       MuleClient client = muleContext.getClient();
-      flowRunner("main").withPayload(inputData).dispatch();
+
+      // Need to return control to test case as soon as message is sent, and not wait for response.
+      muleContext.getSchedulerService().ioScheduler()
+          .submit(() -> flowRunner("main").withPayload(inputData).withExchangePattern(ONE_WAY).dispatch());
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/core/routing/outbound/AggregationTimeoutTestCase.java]: do you still need dispatch here?
